README.md

```diff
@@ -0,0 +1,556 @@
+separate (Flask) web app, apart from CKAN. This was the design because
+the job requires intensive processing to convert every line of the data
+into JSON. However it means more complicated code as info needs to be
+passed between the services in http requests, more for the user to
+set-up and manage - another app config, another apache config, separate
+log files.
+
+XLoader - the job runs in a worker process, in the same app as CKAN, so
+can access the CKAN config, db and logging directly and avoids many HTTP
+calls. This simplification makes sense because the xloader job doesn't
+need to do much processing - mainly it is streaming the CSV file from
+disk into PostgreSQL.
+
+It is still entirely possible to run the XLoader worker on a separate
+server, if that is desired. The worker needs the following:
+
+- A copy of CKAN installed in the same Python virtualenv (but not
+  running).
+- A copy of the CKAN config file.
+- Access to the Redis instance that the running CKAN app uses to store
+  jobs.
+- Access to the database.
+
+You can then run it via ckan jobs worker as below.
+
+### Caveat - column types
+
+Note: With XLoader, all columns are stored in DataStore's database as
+'text' type (whereas DataPusher did some rudimentary type guessing -
+see 'Robustness' above). However once a resource is xloaded, an admin
+can use the resource's Data Dictionary tab to change these types to
+numeric or datestamp and re-load the file. When migrating from
+DataPusher to XLoader you can preserve the types of existing resources
+by using the `migrate_types` command.
+
+There is scope to add functionality for automatically guessing column
+type -offers to contribute this are welcomed.
+
+## Requirements
+
+Compatibility with core CKAN versions:
+
+  | CKAN version   | Compatibility                                                                                                  |
+  | -------------- | -------------------------------------------------------------------------------------------------------------- |
+  | 2.7            | no longer supported (last supported version: 0.12.2)                                                           |
+  | 2.8            | no longer supported (last supported version: 0.12.2)                                                           | 
+  | 2.9            | yes (Python3) (last supported version for Python 2.7: 0.12.2)), Must: `pip install "setuptools>=44.1.0,<71"`   |
+  | 2.10           | yes                                                                                                            |
+  | 2.11           | yes                                                                                                            |
+
+## Installation
+
+To install XLoader:
+
+1.  Activate your CKAN virtual environment, for example:
+
+        . /usr/lib/ckan/default/bin/activate
+
+2.  Install the ckanext-xloader Python package into your virtual
+    environment:
+
+        pip install ckanext-xloader
```

Comment:
can you also give alt method

``pip install git+https://github.com/ckan/ckanext-xloader@master``

Reply:
If you're installing `master` then you probably want a source install:

    pip install -e git+https://github.com/ckan/ckanext-xloader@master